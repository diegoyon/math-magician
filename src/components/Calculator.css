* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-size: 1rem;
}

.grid {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  width: 500px;
  height: 400px;
}

.grid button {
  border: 1px solid gray;
  cursor: pointer;
}

.grid-item-0 {
  grid-column: 1 / -1;
  text-align: end;
  background-color: #858693;
  border: none;
  padding-right: 0.2rem;
  font-size: 1.5rem;
}

.grid-item-0::placeholder {
  color: white;
  padding-right: 0.2rem;
}

.grid-item-17 {
  grid-column: 1/3;
}

.grid button:nth-child(4n+1),
.grid button:nth-child(20) {
  background-color: #f5913e;
}

.grid button:hover {
  filter: brightness(85%);
}
