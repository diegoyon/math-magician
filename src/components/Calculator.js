/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleEvent = (event) => {
    const result = calculate(this.state, event.target.textContent);
    this.setState(result);
  }

  render() {
    const { total, next } = this.state;
    return (
      <div className="grid">
        <input className="grid-item-0" type="input" placeholder={next || total} />

        <button onClick={this.handleEvent} className="grid-item-1" type="button">
          AC
        </button>
        <button onClick={this.handleEvent} className="grid-item-2" type="button">
          +/-
        </button>
        <button onClick={this.handleEvent} className="grid-item-3" type="button">
          %
        </button>
        <button onClick={this.handleEvent} className="grid-item-4" type="button">
          รท
        </button>

        <button onClick={this.handleEvent} className="grid-item-5" type="button">
          7
        </button>
        <button onClick={this.handleEvent} className="grid-item-6" type="button">
          8
        </button>
        <button onClick={this.handleEvent} className="grid-item-7" type="button">
          9
        </button>
        <button onClick={this.handleEvent} className="grid-item-8" type="button">
          x
        </button>

        <button onClick={this.handleEvent} className="grid-item-9" type="button">
          4
        </button>
        <button onClick={this.handleEvent} className="grid-item-10" type="button">
          5
        </button>
        <button onClick={this.handleEvent} className="grid-item-11" type="button">
          6
        </button>
        <button onClick={this.handleEvent} className="grid-item-12" type="button">
          -
        </button>

        <button onClick={this.handleEvent} className="grid-item-13" type="button">
          1
        </button>
        <button onClick={this.handleEvent} className="grid-item-14" type="button">
          2
        </button>
        <button onClick={this.handleEvent} className="grid-item-15" type="button">
          3
        </button>
        <button onClick={this.handleEvent} className="grid-item-16" type="button">
          +
        </button>

        <button onClick={this.handleEvent} className="grid-item-17" type="button">
          0
        </button>
        <button onClick={this.handleEvent} className="grid-item-18" type="button">
          .
        </button>
        <button onClick={this.handleEvent} className="grid-item-19" type="button">
          =
        </button>
      </div>
    );
  }
}

export default Calculator;
