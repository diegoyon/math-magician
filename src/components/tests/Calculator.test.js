import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
