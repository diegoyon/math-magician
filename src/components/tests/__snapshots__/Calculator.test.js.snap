// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="grid"
>
  <div
    className="grid-item-0"
  >
    0
  </div>
  <button
    className="grid-item-1"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="grid-item-2"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="grid-item-3"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="grid-item-4"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="grid-item-5"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="grid-item-6"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="grid-item-7"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="grid-item-8"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="grid-item-9"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="grid-item-10"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="grid-item-11"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="grid-item-12"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="grid-item-13"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="grid-item-14"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="grid-item-15"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="grid-item-16"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="grid-item-17"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="grid-item-18"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="grid-item-19"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
